js/src/components/query/querydiff.tsx

```diff
@@ -495,3 +495,91 @@ export function toDataDiffGrid(
     invalidPKeyCurrent,
   };
 }
+
+export interface QueryDataGridOptions {
+  primaryKeys?: string[];
+  onPrimaryKeyChange?: (primaryKeys: string[]) => void;
+  pinnedColumns?: string[];
+  onPinnedColumnsChange?: (pinnedColumns: string[]) => void;
+}
+
+export function toDataGrid(
```

Comment:
Does it reasonable to reuse the same code as the `QueryResultView.toDataGrid`

Reply:
Merge the PrimaryKey logic into `QueryResultView.toDataGrid` and change to use `QueryResultView.toDataGrid`